lkmpg.tex

```diff
@@ -1502,6 +1502,46 @@ \subsection{Flashing keyboard LEDs}
 Flashing keyboard LEDs can be such a solution: It is an immediate way to attract attention or to display a status condition.
 Keyboard LEDs are present on every hardware, they are always visible, they do not need any setup, and their use is rather simple and non-intrusive, compared to writing to a tty or a file.
 
+From v4.14 to v4.15, the timer API made a series of changes to improve memory safety.
+A buffer overflow in the area of a \cpp|timer_list| structure may be able to overwrite the \cpp|function| and \cpp|data| fields, providing the attacker with a way to use return-object programming (ROP) to call arbitrary functions within the kernel.
+Also, the function prototype of the callback, containing a \cpp|unsigned long| argument, will prevent work from any type checking.
+Furthermore, the function prototype with \cpp|unsigned long| argument may be an obstacle to the \textit{control-flow integrity}, thus, it is better to use a unique prototype to isolate from the others that take an \cpp|unsigned long| argument.
```

Comment:
Split the sentence starting "thus" in the next iine.

Reply:
Improve the writing: "isolate from the others that take an \cpp|unsigned long| argument" looks weird.